kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -163,6 +193,53 @@ public Transaction build(Engine engine) {
       }
     }
 
+    /* --------------- Handle Domain Metadata ------------ */
```

Comment:
I can move this code out to a helper if desired / I think it may conflict with @vkorukanti TF changes somewhat

Reply:
This needs revisit depending on which PR goes in first.

Reply:
Going to take a look at the TF PR now and then I will come back to this based on thoughts / which PR is going in